subprojects/base-services/src/main/java/org/gradle/internal/classloader/ClassLoaderFactory.java

```diff
@@ -26,7 +26,7 @@
     /**
      * Creates a ClassLoader implementation which has only the classes from the specified URIs and the Java API visible.
      */
-    ClassLoader createIsolatedClassLoader(ClassPath classPath);
+    ClassLoader createIsolatedClassLoader(String name, ClassPath classPath);
```

Comment:
I'm pretty sure this will break some people builds/plugins.
I can live with that given it's internal and aimed at 5.0.